content/en/references/init-hooks.md

```diff
@@ -151,6 +151,76 @@ DOCKER_FLAGS='-v /path/to/init-aws.sh:/etc/localstack/init/ready.d/init-aws.sh'
 
 Another use for init hooks can be seen when [adding custom TLS certificates to LocalStack]({{< ref "custom-tls-certificates#custom-tls-certificates-with-init-hooks" >}}).
 
+### Terraform configuration files as init hooks
+
+Running Terraform configuration files as init hooks is in itself its own category, because it requires the installation of a special extension.
+For more information on how to manage [LocalStack extensions](({{< ref "user-guide/extensions/" >}})), please refer to the dedicated documentation page,
+and for more details on running init hooks in development mode, you can check out the [extension repository description](https://github.com/localstack/localstack-extensions/tree/main/terraform-init).
+
+##### Usage
+
+Start LocalStack with **`EXTENSION_AUTO_INSTALL="localstack-extension-terraform-init"`**.
+Mount a **`main.tf`** file into **`/etc/localstack/init/ready.d`**
+When LocalStack starts up, it will install the extension, which in turn installs Terraform and [`tflocal`](https://github.com/localstack/terraform-local) into the container.
+If one of the init stage directories contain a `main.tf` file, the extension will run `tflocal init` and `tflocal apply` on that directory.
```

Comment:
The usage example should demonstrate two scenarios — using Docker Compose and `localstack` CLI right in this part. I would say that with `localstack` CLI, it should be good to demonstrate the `localstack extensions install....`